megamek/src/megamek/common/enums/BuildingType.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2024 - The MegaMek Team. All Rights Reserved.
+ *
+ * This file is part of MegaMek.
+ *
+ * MegaMek is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MegaMek is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MegaMek. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package megamek.common.enums;
+
+import java.util.Arrays;
+import java.util.ResourceBundle;
+
+import megamek.MegaMek;
+
+public enum BuildingType {
+    UNKNOWN("BuildingType.UNKNOWN.text", -1, -1),
+    LIGHT("BuildingType.LIGHT.text", 1, 15),
+    MEDIUM("BuildingType.MEDIUM.text", 2, 40),
+    HEAVY("BuildingType.HEAVY.text", 3, 90),
+    HARDENED("BuildingType.HARDENED.text", 4, 120),
+    WALL("BuildingType.WALL.text", 5, 120);
+    
+    private final String name;
+    private final int type;
+    private final int defaultCF;
+    
+    private BuildingType(String name, int type, int defaultCF) {
+        final ResourceBundle resources = ResourceBundle.getBundle("megamek.common.messages",
+                MegaMek.getMMOptions().getLocale());
```

Comment:
Can this be moved to a class member?
Or is there no danger of map draw slowdowns because we only instantiate this once per Building Type, early on?

Reply:
because this is an enum it will load only once